### PR TITLE
Bump versions

### PR DESCRIPTION
**Pull request by oerc0122**
Ensures CI builds work again by:

* Refactor/clean `create_mltbx.py`
* Updates versions of CI "actions" to latest (using Node.js 20)
* Updates code to account for changes in Matlab between R2021b and R2024a
* Updates ReadTheDocs configuration.

### CODE REVIEW

File: update_module_versions.py [hunk 1]
```diff
@@ -4,11 +4,12 @@
 def get_module_versions():
     # gets the required module versions from `min_requirements.txt` file
     curdir = os.path.dirname(os.path.abspath(__file__))
-    req_file = os.path.join(
-        curdir, 'euphonic_sqw_models', 'min_requirements.txt')
-    if not os.path.isfile(req_file):
-        from update_dependencies import update_submodules
-        update_submodules('euphonic_sqw_models')
+    for submodule in [['light_python_wrapper', '+light_python_wrapper'],
```
Comment (oerc0122) [inline update_module_versions.py:7]: Urgh, `os.path`...

**Review by oerc0122 — COMMENTED** [review]
Besides my gripes, all looks good

File: update_module_versions.py (hunk 1 above)
Comment (mducle) [reply update_module_versions.py:7]: Nothing wrong with that... I don't like to divide my paths...

Comment (oerc0122) [reply update_module_versions.py:7]:
```python
>>> Path() / 0
ZeroDivisionError: division by zero
```
oh no!

**Pull request merged**